platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/common/DependencyCollectorFunctionExtractorAndRuntimeResolver.kt

```diff
@@ -75,38 +78,73 @@ class DependencyCollectorFunctionExtractorAndRuntimeResolver(
 
     private
     fun expandToOverloads(host: SchemaBuildingHost, produceDeclaration: (DataParameter) -> DependencyCollectorDeclaration) =
-        listOf(gavDependencyParam, projectDependencyParam).map {
+        listOf(gavDependencyParam, dependencyParam).map {
             val param = it(host)
             produceDeclaration(param)
         }
 
     override fun memberFunctions(host: SchemaBuildingHost, kClass: KClass<*>, preIndex: DataSchemaBuilder.PreIndex): Iterable<SchemaMemberFunction> {
```

Comment:
💅 Given the growth of this `memberFunctions` implementation, consider extracting the parts working with the dependency collectors and the platform modifier into functions to improve comprehensibility.

Reply:
Yeah, that's probably a good idea.  I'll extract some functions before I merge this.

Funny story: I thought about doing this when cleaning up my changes for this PR, then saw IntelliJ rate this function as "low complexity (33%)" and didn't.